net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/config/ConfigProperty.java

```diff
@@ -46,10 +46,9 @@ public class ConfigProperty implements Comparable<ConfigProperty>, Cloneable {
    *          the property meta data
    */
   public ConfigProperty(ConfigPropertyMetadata metaData) {
-
     this(metaData.getName(), metaData.getOverrideDefault() != null ? metaData.getOverrideDefault()
             : metaData.getDefaultValue());
-    setMetaData(metaData);
+    this.metaData = metaData;
```

Comment:
I'm not a friend of inlining setters. The moment someone adds an additional line in a setter, like a precondition, a null check etc., all the inlined duplications of it are wrong, because they miss that same change. Any reason why you are doing that?

Reply:
I think there's a difference between inlining setters and moving away from using setters in constructors, which is what I'm doing here.
Usually, in the body of a method, you should be able to assume you have a fully constructed object. Calling methods in constructors is therefore not ideal.
If I had validation logic I wanted applied in the constructor, and it was worth sharing it with the setter, I'd make a private method that performs this validation.
I'm not worried about someone "miss[ing] that same change", because it would be bad to assume that all constructors are all calling the setters: it's not generally the case.

The alternative to this is just to finalize the setters used in constructors, which I've done once in this PR, but I think not using setters in constructors at all is a better solution.